### ISSUE TITLE
Errors when eager loading entries

### ISSUE THREAD

**Issue by michaelhue** [vaersaagod/seomate#19]
## Problem

SEOMate throws an error when using eager-loading on asset/relation fields for the page's current `entry`. The problem is that SEOMate expects all relation fields to be an `ElementQuery` and tries to call `one()` or `all()` on them, which doesn't work if the fields are already eager loaded.

### SEOMate config

```php
<?php
return [
    'fieldProfiles' => [
	'default' => [
		'image' => ['image'],
	],
    ],
];
```

### Template

```twig
{% do eagerLoad(entry, ['image']) %}
```

### Result

```
Error
Call to a member function all() on array
in /src/vendor/vaersaagod/seomate/src/helpers/SEOMateHelper.php at line 162
```

(see below for stack trace)

## Possible solution

The problematic line in this case:

```php
$assets = $scope[$handle]->all() ?? null;
```

A possible solution would be to add a type check before trying to execute `ElementQuery` queries, something like this:

```php
$elements = $scope[$handle];
$assets = ($elements instanceof ElementQuery) ? $elements->all() : $elements;
```

## Full stack trace
```
Error: Call to a member function all() on array in /src/vendor/vaersaagod/seomate/src/helpers/SEOMateHelper.php:162
Stack trace:
#0 /src/vendor/vaersaagod/seomate/src/services/MetaService.php(175): vaersaagod\seomate\helpers\SEOMateHelper::getPropertyDataByScopeAndHandle(Object(craft\elements\Entry), 'image', 'image')
#1 /src/vendor/vaersaagod/seomate/src/services/MetaService.php(158): vaersaagod\seomate\services\MetaService->getElementPropertyDataByFields(Object(craft\elements\Entry), 'image', Array)
#2 /src/vendor/vaersaagod/seomate/src/services/MetaService.php(139): vaersaagod\seomate\services\MetaService->generateElementMetaByProfile(Object(craft\elements\Entry), Array)
#3 /src/vendor/vaersaagod/seomate/src/services/MetaService.php(63): vaersaagod\seomate\services\MetaService->getElementMeta(Object(craft\elements\Entry), NULL)
#4 /src/vendor/vaersaagod/seomate/src/SEOMate.php(237): vaersaagod\seomate\services\MetaService->getContextMeta(Array)
#5 /src/vendor/craftcms/cms/src/web/View.php(1276): vaersaagod\seomate\SEOMate->onRegisterMetaHook(Array)
#6 /src/storage/runtime/compiled_templates/4b/4b2c44a6015cefb6a4c05340cf59c0e37485a7efceebe7b867cbcdcadac4d37a.php(50): craft\web\View->invokeHook('seomateMeta', Array)
#7 /src/vendor/twig/twig/src/Template.php(407): __TwigTemplate_87324b2e0d31f3e9d681217f96a386402061434a0e90e52237808e20992535a5->doDisplay(Array, Array)
#8 /src/vendor/craftcms/cms/src/web/twig/Template.php(52): Twig\Template->displayWithErrorHandling(Array, Array)
#9 /src/vendor/twig/twig/src/Template.php(380): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#10 /src/vendor/craftcms/cms/src/web/twig/Template.php(34): Twig\Template->display(Array, Array)
#11 /src/storage/runtime/compiled_templates/97/970eb2c72f5b817deb7c4f4713a40dbfc73ed60f3fc5405a0c4edff08a2c4314.php(56): craft\web\twig\Template->display(Array, Array)
#12 /src/vendor/twig/twig/src/Template.php(407): __TwigTemplate_b1093385c726217cb623e102fcae78a38f922fae40477b3dc0cc549c282a21fe->doDisplay(Array, Array)
#13 /src/vendor/craftcms/cms/src/web/twig/Template.php(52): Twig\Template->displayWithErrorHandling(Array, Array)
#14 /src/vendor/twig/twig/src/Template.php(380): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#15 /src/vendor/craftcms/cms/src/web/twig/Template.php(34): Twig\Template->display(Array, Array)
#16 /src/vendor/twig/twig/src/Template.php(392): craft\web\twig\Template->display(Array)
#17 /src/vendor/twig/twig/src/Extension/CoreExtension.php(1209): Twig\Template->render(Array)
#18 /src/storage/runtime/compiled_templates/d3/d32439cd11b8d1c5fe2aa5b1980b0612d63596d2994d8364f9dd21afb1c2b72f.php(39): twig_include(Object(craft\web\twig\Environment), Array, '_publications/p...', Array, true)
#19 /src/vendor/twig/twig/src/Template.php(407): __TwigTemplate_a263b6b643c872cb32027411f5cd2c2a078d124197ba1fd6f3a99b6d49e8cded->doDisplay(Array, Array)
#20 /src/vendor/craftcms/cms/src/web/twig/Template.php(52): Twig\Template->displayWithErrorHandling(Array, Array)
#21 /src/vendor/twig/twig/src/Template.php(380): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#22 /src/vendor/craftcms/cms/src/web/twig/Template.php(34): Twig\Template->display(Array, Array)
#23 /src/vendor/twig/twig/src/Template.php(392): craft\web\twig\Template->display(Array)
#24 /src/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#25 /src/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#26 /src/vendor/craftcms/cms/src/web/View.php(344): Twig\Environment->render('_publications/e...', Array)
#27 /src/vendor/craftcms/cms/src/web/View.php(394): craft\web\View->renderTemplate('_publications/e...', Array)
#28 /src/vendor/craftcms/cms/src/web/Controller.php(243): craft\web\View->renderPageTemplate('_publications/e...', Array)
#29 /src/vendor/craftcms/cms/src/controllers/TemplatesController.php(95): craft\web\Controller->renderTemplate('_publications/e...', Array)
#30 [internal function]: craft\controllers\TemplatesController->actionRender('_publications/e...', Array)
#31 /src/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#32 /src/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#33 /src/vendor/craftcms/cms/src/web/Controller.php(187): yii\base\Controller->runAction('render', Array)
#34 /src/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('render', Array)
#35 /src/vendor/craftcms/cms/src/web/Application.php(299): yii\base\Module->runAction('templates/rende...', Array)
#36 /src/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#37 /src/vendor/craftcms/cms/src/web/Application.php(284): yii\web\Application->handleRequest(Object(craft\web\Request))
#38 /src/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#39 /src/web/index.php(21): yii\base\Application->run()
#40 {main}
```

**Comment by mmikkel** [comment]
Hi @michaelhue, thanks for reporting. I'll look into adding support for eager-loaded relational fields.

**Comment by mmikkel** [comment]
Fixed in [SEOMate 1.1.10](https://github.com/vaersaagod/seomate/releases/tag/v1.1.10)
